specs/cli/plugin-add.md

```diff
@@ -0,0 +1,163 @@
+| `--type \| -t` | Yes | openai | The target type of plugin for the generated output files. Accepts multiple values. Possible values are `openai` and `apimanifest`. Defaults to `apimanifest`| Yes |
+| `--overlayDirectory \| --od` | No | ./overlays/plugins/{plugin-name}/overlay.yaml | The location of the overlay file in JSON or YAML format to be used to generate the plugin. [Overlay](https://github.com/OAI/Overlay-Specification/blob/main/versions/1.0.0.md) defines a way of creating documents that contain additional information to be merged with an OpenAPI description. Defaults to no value which uses the OpenAPI description as it is. | Yes, without its value |
+| `--skip-generation \| --sg` | No | true | When specified, the generation would be skipped. Defaults to false. | Yes |
+| `--output \| -o` | No | ./generated/plugins/github | The output directory or file path for the generated output files. This is relative to the location of `workspace.json`. Defaults to `./output`. | Yes, without its value |
+
+> [!NOTE] 
+> It is not required to use the CLI to add new plugins. It is possible to add a new plugin by adding a new entry in the `plugins` section of the `workspace.json` file. See the [workspace.json schema](../schemas/workspace.json) for more information. Using `kiota plugin generate --plugin-name myPlugin` would be required to generate the plugins.
+
+## Using `kiota plugin add`
+
+```bash
+kiota plugin add --plugins-name "GitHub" --openapi "https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json" --include-path "/repos/{owner}/{repo}" --type openai, apimanifest --output "./generated/plugins/github"
+```
+
+_The resulting `workspace.json` file will look like this:_
+
+```jsonc
+{
+  "version": "1.0.0",
+  "clients": {...}, //if any
+  "plugins": {
+    "GitHub": {
+      "descriptionLocation": "https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json",
+      "includePatterns": ["/repos/{owner}/{repo}"],
+      "excludePatterns": [],
+      "type": "openai",
+      "outputDirectory": "./generated/plugins/github",
+      "overlayDirectory": "./overlays/plugins/github/overlay.yaml"
+    }
+  }
+}
+```
+
+_The resulting `github-openai.json` file will look like this:_
+
+```jsonc
+{
+    "schema_version": "v1",
+    "name_for_human": "GitHub Repository Plugin",
+    "name_for_model": "github",
+    "description_for_human": "Manage GitHub repositories.",
+    "description_for_model": "Help the user with managing a GitHub repositories. You can view, update and remove repositories.",
+    "auth": {
+        "type": "none"
+    },
+    "api": {
+        "type": "openapi",
+        "url": "./sliced-openapi-github.json"
+    },
+    "logo_url": "https://example.com/logo.png",
+    "contact_email": "githubsupport@example.com",
+    "legal_info_url": "http://www.example.com/view-plugin-information"
```

Comment:
where are we supposed to source that information from ? the OpenAPI description?

Reply:
Yes. The source should be the OpenAPI description. OpenAPI.NET lib has this logic already implemented [from OpenAPI to API Manifest](https://github.com/microsoft/OpenApi.ApiManifest/blob/ee957a55c30bc09b45d81fcff9a2ffa18ead8393/docs/OpenApiToApiManifestMapping.md) that is used later to convert [API Manifest to OpenAI](https://github.com/microsoft/OpenApi.ApiManifest/blob/46b4974f17be8fb85c84d75b75e7244b5fa81404/src/lib/TypeExtensions/ApiManifestDocumentExtensions.cs#L21) and we should use the same logic to plugin generation in Kiota.

Reply:
Can we have references to those documents in this spec?

Reply:
Done.